src/services/SnapshotService.ts

```diff
@@ -3,20 +3,62 @@ import isNumber from 'lodash/isNumber'
 
 import { SnapshotApi, SnapshotReceipt } from '../clients/SnapshotApi'
 import { SnapshotGraphql } from '../clients/SnapshotGraphql'
-import { DetailedScores, SnapshotProposal, SnapshotVote, VpDistribution } from '../clients/SnapshotGraphqlTypes'
+import {
+  DetailedScores,
+  ServiceHealth,
+  SnapshotProposal,
+  SnapshotStatus,
+  SnapshotVote,
+  VpDistribution,
+} from '../clients/SnapshotTypes'
 import * as templates from '../entities/Proposal/templates'
 import { proposalUrl, snapshotProposalUrl } from '../entities/Proposal/utils'
 import { SNAPSHOT_SPACE } from '../entities/Snapshot/constants'
 import { isSameAddress } from '../entities/Snapshot/utils'
 import { inBackground } from '../helpers'
 import { Avatar } from '../utils/Catalyst/types'
+import { ErrorCategory } from '../utils/errorCategories'
 
+import CacheService from './CacheService'
+import { ErrorService } from './ErrorService'
 import { ProposalInCreation, ProposalLifespan } from './ProposalService'
 import RpcService from './RpcService'
 
 const DELEGATION_STRATEGY_NAME = 'delegation'
+const SLOW_RESPONSE_TIME_THRESHOLD_IN_MS = 5000 // 5 seconds
+const PING_INTERVAL_IN_SECONDS = 60 // 1 minute
 
 export class SnapshotService {
+  public static async getStatus(): Promise<SnapshotStatus | undefined> {
+    try {
+      const cachedStatus = CacheService.get<SnapshotStatus>('snapshotStatus')
+      if (cachedStatus) {
+        return cachedStatus
+      }
+
+      let health = ServiceHealth.Normal
+      const responseTime = await SnapshotGraphql.get().ping()
+      if (responseTime === -1) {
+        health = ServiceHealth.Failing
+      } else if (responseTime > SLOW_RESPONSE_TIME_THRESHOLD_IN_MS) {
+        health = ServiceHealth.Slow
+      }
+
+      const snapshotStatus = { health, responseTime }
+      this.saveStatusInCache(snapshotStatus)
```

Comment:
If we are not reusing this method I'd call CacheService directly here as the method is a one-liner.